From: Takashi Iwai <tiwai@suse.de>
Subject: kABI workaround for hid_debug_list kfifo changes
Patch-mainline: Never, SLE15 only
References: CVE-2019-3819, bsc#1123161

The patch
  patches.drivers/HID-debug-fix-the-ring-buffer-implementation.patch
leads to kABI incompatibility due to the inclusion of linux/kfifo.h.

Wrap it with ifdef, and remove the superfluous include as well.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/hid/hid-debug.c   |    1 -
 include/linux/hid-debug.h |    2 ++
 2 files changed, 2 insertions(+), 1 deletion(-)

--- a/include/linux/hid-debug.h
+++ b/include/linux/hid-debug.h
@@ -24,7 +24,9 @@
 
 #ifdef CONFIG_DEBUG_FS
 
+#ifndef __GENKSYMS__
 #include <linux/kfifo.h>
+#endif
 
 #define HID_DEBUG_BUFSIZE 512
 #define HID_DEBUG_FIFOSIZE 512
--- a/drivers/hid/hid-debug.c
+++ b/drivers/hid/hid-debug.c
@@ -30,7 +30,6 @@
 
 #include <linux/debugfs.h>
 #include <linux/seq_file.h>
-#include <linux/kfifo.h>
 #include <linux/sched/signal.h>
 #include <linux/export.h>
 #include <linux/slab.h>
